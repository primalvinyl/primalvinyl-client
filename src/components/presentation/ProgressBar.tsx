import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
    padding: 0;
    margin: 0;
    width: 100%;

    div.ldsEllipsis {
        position: relative;
        width: 80px;
        height: 10px;
        margin: 0 auto;

        div {
            position: absolute;
            top: 0;
            width: 10px;
            height: 10px;
            border-radius: 50%;
            background: ${props => props.color};
            animation-timing-function: cubic-bezier(0, 1, 1, 0);
        }

        div:nth-child(1) {
            left: 8px;
            animation: ldsEllipsis1 0.6s infinite;
        }

        div:nth-child(2) {
            left: 8px;
            animation: ldsEllipsis2 0.6s infinite;
        }

        div:nth-child(3) {
            left: 32px;
            animation: ldsEllipsis2 0.6s infinite;
        }

        div:nth-child(4) {
            left: 56px;
            animation: ldsEllipsis3 0.6s infinite;
        }
    }

    @keyframes ldsEllipsis1 {
        0% {
            transform: scale(0);
        }
        100% {
            transform: scale(1);
        }
    }

    @keyframes ldsEllipsis3 {
        0% {
            transform: scale(1);
        }
        100% {
            transform: scale(0);
        }
    }

    @keyframes ldsEllipsis2 {
        0% {
            transform: translate(0, 0);
        }
        100% {
            transform: translate(24px, 0);
        }
    }
`;

const ProgressBar = ({ color }: ProgressBarProps): React.ReactElement => {
    return (
        <Wrapper color={color}>
            <div className="ldsEllipsis" role="progressbar">
                <div></div><div></div><div></div><div></div>
            </div>
        </Wrapper>
    );
};

type ProgressBarProps = {
    readonly color?: string;
};

ProgressBar.defaultProps = {
    color: 'black'
};

export default ProgressBar;
