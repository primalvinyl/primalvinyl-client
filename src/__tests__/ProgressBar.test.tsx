import * as React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import ProgressBar from '../components/presentation/ProgressBar';

describe('ProgressBar', () => {
    it('renders', () => {
        const myWrapper = shallow(<ProgressBar />);
        expect(myWrapper.exists());
    });
    
    it('matches snapshot', () => {
        const component = Renderer.create(<ProgressBar />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});