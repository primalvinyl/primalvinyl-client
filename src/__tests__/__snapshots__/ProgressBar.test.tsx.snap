// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar matches snapshot 1`] = `
<div
  className="sc-bdfBwQ cAIpLZ"
  color="black"
>
  <div
    className="ldsEllipsis"
    role="progressbar"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
